cime_config/buildcpp

```diff
@@ -79,115 +83,125 @@ def buildcpp(case):
-         case.set_value("CICE_DECOMPTYPE", "roundrobin")
 
-    if cice_mode == 'prescribed':
-        ntr_aero = 0
-    else:
-        ntr_aero = 3
-    ntr_aero = set_nondefault_cpp(cice_config_opts, "ntr_aero", ntr_aero)
-    case.set_value("ICE_NAERO", ntr_aero)
-
-    # set number of bgc tracers (valid values are 0->10)
-    if cice_mode == 'prescribed':
-        nbgclyr = 0
-    else:
-        nbgclyr = 3
-    nbgclyr = set_nondefault_cpp(cice_config_opts, "nbgclyr", nbgclyr)
-
-    # set isotope tracer
-    ntr_iso = 0
-    ntr_iso = set_nondefault_cpp(cice_config_opts, "ntr_iso", ntr_iso)
-
-    # set age tracer
-    trage = 1
-    trage = set_nondefault_cpp(cice_config_opts, "trage", trage)
-
-    # set first year ice tracer (valid values are 0,1)
-    trfy = 1
-    trfy = set_nondefault_cpp(cice_config_opts, "trfy", trfy)
-
-    # set pond tracer (valid values are 0,1)
-    trpnd = 1
-    trpnd = set_nondefault_cpp(cice_config_opts, "trpnd", trpnd)
-
-    # set level ice tracer (valid values are 0,1)
-    trlvl = 1
-    trlvl = set_nondefault_cpp(cice_config_opts, "trlvl", trlvl)
-
-    # set brine tracer (valid values are 0,1)
-    trbri = 0
-    trbri = set_nondefault_cpp(cice_config_opts, "trbri", trbri)
-
-    # set skeletal layer tracer (valid values are 0,1)
-    trbgcs = 0
-    trbgcs = set_nondefault_cpp(cice_config_opts, "trbgcs", trbgcs)
-
-    # set number of ice layers
-    if phys == "cice4":
-        nicelyr = 4
-    elif "ar9v" in hgrid:
-        nicelyr = 7
-    else:
-        nicelyr = 8
-    case.set_value("ICE_NILYR", nicelyr)
-
-    # set number of snow layers
-    if phys == "cice4":
-        nsnwlyr = 1
-    else:
-        nsnwlyr = 3
-    case.set_value("ICE_NSLYR", nsnwlyr)
-
-    # set number of ice categories
-    # NOTE that ICE_NCAT is used by both cice and pop - but is set by cice
-    # and as a result it is assumed that the cice buildcpp is called
-    # BEFORE the pop buildcpp. This order is set by the xml variable
-    # COMP_CLASSES in the driver config_component.xml file
-    if cice_mode == 'prescribed':
-        ncat = 1
-    else:
-        ncat = 5
-    ncat = set_nondefault_cpp(cice_config_opts, "ncat", ncat)
-    case.set_value("ICE_NCAT",ncat)
-    logger.debug("cice: number of ice categories (ncat) is %s" %ncat)
-
-    # set decomposition block sizes
-    cice_blckx   = case.get_value("CICE_BLCKX")
-    cice_blcky   = case.get_value("CICE_BLCKY")
-    cice_mxblcks = case.get_value("CICE_MXBLCKS")
-
-    cice_cppdefs = " -DCESMCOUPLED -Dncdf -DNUMIN=11 -DNUMAX=99 " \
-                   " -DNICECAT=%s -DNXGLOB=%s -DNYGLOB=%s -DNTRAERO=%s -DNTRISO=%s" \
-                   " -DNBGCLYR=%s -DNICELYR=%s -DNSNWLYR=%s"  \
-                   " -DTRAGE=%s -DTRFY=%s -DTRLVL=%s -DTRPND=%s -DTRBRI=%s -DTRBGCS=%s" \
-                   %(ncat,nx,ny,ntr_aero,ntr_iso,nbgclyr,nicelyr,nsnwlyr,trage,trfy,trlvl,trpnd,trbri,trbgcs)
+# tcraig, moved to namelist_definition_cice.xml
```

Comment:
If the following are namelist settings - this logic should be moved to buildnml - and removed from buildcpp. Please remove the commented out code - since its clear from the PR code what has been removed - and this will make buildcpp much easier to understand.

Reply:
As I noted above, I continue to work on the cleanup.  I'm committing things as I make progress.  I think we all recognize there is still work to do.

Reply:
Sounds fine.  I committing my changes to my branch as you suggested.  I created PRs as you suggested.  I think it's great that others are looking at what changes are coming in as I make progress.  I encourage others to provide feedback, I don't really have a good handle on all aspects of these scripts.  There is still work to do before this is done, and it might not "all" get done just in this PR.  We can merge the PR whenever folks want.